sops.go

```diff
@@ -103,40 +111,32 @@ type Tree struct {
 	Metadata Metadata
 }
 
-// TrimTreePathComponent trimps a tree path component so that it's a valid tree key
-func TrimTreePathComponent(component string) (string, error) {
-	if component[len(component)-1] != ']' {
-		return "", fmt.Errorf("Invalid component")
-	}
-	component = component[:len(component)-1]
-	component = strings.Replace(component, `"`, "", 2)
-	component = strings.Replace(component, `'`, "", 2)
-	return component, nil
-}
-
-// Truncate truncates the tree following Python dictionary access syntax, for example, ["foo"][2].
-func (tree TreeBranch) Truncate(path string) (interface{}, error) {
-	components := strings.Split(path, "[")
+// Truncate truncates the tree to the path specified
+func (tree TreeBranch) Truncate(path []interface{}) (interface{}, error) {
```

Comment:
Reasoning behind this change: parsing of the component string should take place in the `cmd` package, not in the `sops` package.